pyiges/iges.py

```diff
@@ -229,6 +229,26 @@ def from_pointer(self, ptr):
         """Return an iges object according to an iges pointer"""
         return self[self._pointers[ptr]]
 
+    @staticmethod
+    def _parse_separators_from_first_global_line(line):
+        if line[0] == ',':
+            a = ','
+            if line[1] == a:
+                b = ';'
+            elif line[1:3] == '1H':
+                b = line[3]
+            else:
+                raise Exception('Invalid Header format')
+        elif line[0:2] == '1H':
+            a = line[2]
+            if line[4:6] == '1H':
+                b = line[6]
+            elif line[3] == a:
+                b = ';'
+            else:
+                raise Exception('Invalid Header format')
+        return a, b
+
```

Comment:
Note that the new method is fully covered by the new tests

![grafik](https://user-images.githubusercontent.com/17496175/232623097-e1303f2a-a042-48c4-b9d6-e0799472a1de.png)

Reply:
I'd go with `RuntimeError` and potentially consider adding a `pragma: no cover` to these lines.

Reply:
Used `RuntimeError`, but instead of pragma, I covered the exception cases also and fixed the missing case when the `elif` in line 241 (screenshot) evals to `False`.

Now every line that has been touched is covered except of this very unimportant one:

![grafik](https://user-images.githubusercontent.com/17496175/232892480-0cbbaa37-05bf-49ff-8564-c6b457051415.png)

Speaking of this, note that the except-pass there is not speculative. If the data is incorrupt, it is unlikely but possible that one entity object gets initialized twice, hence fails twice and is attempted to be discarded twice.